### ISSUE TITLE
Unable to locate package php8.2-dev

### ISSUE THREAD

**Issue by withinboredom** [shivammathur/setup-php#880]
**Describe the bug**

I am running on self-hosted and getting an error message when trying to setup php. I reran the script with `@verbose`:

```
Reading package lists...


WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
Package php8.2-cgi is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-cli is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-curl is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-fpm is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-intl is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-mbstring is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-mysql is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-opcache is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-pgsql is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-xml is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package php8.2-zip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'php8.2-cgi' has no installation candidate
E: Package 'php8.2-cli' has no installation candidate
E: Package 'php8.2-curl' has no installation candidate
E: Unable to locate package php8.2-dev
E: Couldn't find any package by glob 'php8.2-dev'
E: Couldn't find any package by regex 'php8.2-dev'
E: Package 'php8.2-fpm' has no installation candidate
E: Package 'php8.2-intl' has no installation candidate
E: Package 'php8.2-mbstring' has no installation candidate
E: Package 'php8.2-mysql' has no installation candidate
E: Package 'php8.2-opcache' has no installation candidate
E: Package 'php8.2-pgsql' has no installation candidate
E: Package 'php8.2-xml' has no installation candidate
E: Package 'php8.2-zip' has no installation candidate
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package php8.2-dev
E: Couldn't find any package by glob 'php8.2-dev'
E: Couldn't find any package by regex 'php8.2-dev'

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package php8.2-dev
E: Couldn't find any package by glob 'php8.2-dev'
E: Couldn't find any package by regex 'php8.2-dev'
/home/runner/_work/_actions/shivammathur/setup-php/verbose/src/scripts/../scripts/extensions/add_extensions.sh: line 13: php: command not found
/home/runner/_work/_actions/shivammathur/setup-php/verbose/src/scripts/../scripts/extensions/add_extensions.sh: line 16: php: command not found
/home/runner/_work/_actions/shivammathur/setup-php/verbose/src/scripts/../scripts/extensions/add_extensions.sh: line 24: ext_dir: parameter null or not set
```

My runner image is essentially the github actions builder provided by GitHub: https://github.com/bottledcode/sane-builder/blob/latest/Dockerfile

**Version**
- [ ] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [ ] GitHub Hosted
- [x] Self Hosted

**Operating systems**

**PHP versions**

8.2 only, 8.3 and 8.4 work

**To Reproduce**

```
        uses: shivammathur/setup-php@verbose
        with:
          php-version: ${{ matrix.php-versions }}
          ini-file: development
          coverage: none
          tools: none
        env:
          phpts: ts
          debug: true
          runner: self-hosted
```

**Expected behavior**

PHP 8.2 to be installed

**Screenshots/Logs**

Can view output here: https://github.com/bottledcode/frankenphp/actions/runs/11649872124/job/32437911761

**Additional context**

Experimenting with self-hosted runners to get better build speeds for our project.

**Are you willing to submit a PR?**

Not sure how helpful I can be here, but if I can, I will.

**Comment by SagarGi** [comment]
hi @withinboredom i have also got similar problem. https://github.com/nextcloud/integration_openproject/actions/runs/11966137936/job/33361269014 .
Works on ubuntu` 20 LTS` and `22 LTS` but fails for `24 LTS` @shivammathur

**Comment by shivammathur** [comment]
@SagarGi Can you test once with the develop branch, and let me know if that works

**Comment by shivammathur** [comment]
Fixed in https://github.com/shivammathur/setup-php/commit/51383ecefe4db3cb1a7f075b9c1b172e8ee6576e

**Comment by shivammathur** [comment]
Fix released in [`2.32.0`](https://github.com/shivammathur/setup-php/releases/tag/2.32.0).
